lib/oppen/pretty_printer.rb

```diff
@@ -0,0 +1,254 @@
+# frozen_string_literal: true
+
+require 'stringio'
+
+require_relative 'scan_stack'
+require_relative 'print_stack'
+
+#
+# Oppen module.
+#
+module Oppen
+  #
+  # PrettyPrinter class responsible for scanning tokens and passing
+  # them to the printer.
+  #
+  class PrettyPrinter
+    #
+    # Initialization of the needed attributes.
+    # Method called PrettyPrintInit in the original paper.
+    #
+    # @param [integer] line_width  maximum line width desired.
+    #
+    def initialize(line_width)
+      # Maximum size if the stacks
+      n = 3 * line_width
+
+      # Array of token initially filled with nil that represents our token buffer.
+      # Originaly called token in the papper, changed to tokens here for better
+      # representation of its content.
+      @tokens = Array.new n
+
+      # Array of integers initially filled with nil that represents our token buffer.
+      # Attribute possess the same name in the original paper.
+      @size = Array.new n
+
+      # Stack of integer that represents a stack where we save potential breaking positions.
+      # Attribute possess the same name in the original paper.
+      @scan_stack = ScanStack.new n
+
+      # Integers that store the total number of spaces needed to print all elements of the
+      # buffer from its start until left/right.
+      # Both attributes possess the same name in the original paper.
+      @left_total = @right_total = 1
```

Comment:
The description is not that clear, what's `left/right` in this context?
You're talking about the left/right of the ring buffer ? If so, simplify the description and make it clearer.